src/kvstore.c

```diff
@@ -155,11 +155,14 @@ static void cumulativeKeyCountAdd(kvstore *kvs, int didx, long delta) {
     }
 }
 
-static void createDictIfNeeded(kvstore *kvs, int didx) {
-    if (kvstoreGetDict(kvs, didx))
-        return;
+/* Create the dict if it does not exist and return it. */
+static dict *createDictIfNeeded(kvstore *kvs, int didx) {
+    dict *d = kvstoreGetDict(kvs, didx);
+    if (d) return d;
+
     kvs->dicts[didx] = dictCreate(&kvs->dtype);
     kvs->allocated_dicts++;
+    return kvs->dicts[didx];
```

Comment:
a thought while researching this comment https://github.com/redis/redis/pull/12822#pullrequestreview-1855609585, here, do we need to check the KVSTORE_ALLOCATE_DICTS_ON_DEMAND flag?

Reply:
i don't think so.
we need a dict either way (we can't proceed without it).
if it's not an "ON DEMAND" then it means the dict was already created at startup, and we're not expecting NULL.
am i missing anything?

Reply:
yes, that is right. i mean, only ON DEMAND we will create the dict via this ifNeeded function. If it is not an ON DEMAND, it will be created in startup, if it is no created, it is a bug and then we should catch it (either return NULL or do an assert). but i might be thinking too much, just raise this comment.

Reply:
i think that it's easy enough to realize that this can't happen, and there's no need for any explicit check or assertion.